### PR TITLE
EntityTasks

### PR DESCRIPTION
**Pull request by applejuiceyy**
entity tasks or something
image

### CODE REVIEW

**Comment by KitCat962** [comment]
can you give the entity rendertask a context for the render event?

**Comment by applejuiceyy** [comment]
I have the slight idea that they shouldn't render through figura, I remember enabling it for player skulls in item tasks and ending up opening a giant can of worms

**Comment by KitCat962** [comment]
can I suggest adding a function to allow generating an Entity copy from an existing entity?

Also, how do you use the setNbt function? What is the format of the nbt it is expecting?

**Comment by applejuiceyy** [comment]
just the usual nbt string, like the summon command

**Comment by KitCat962** [comment]
are you sure about that?
errors with or without the `{}`

image

also yea vm error bad

**Comment by PoolloverNathan** [comment]
> are you sure about that?
> errors with or without the `{}`
> 
> image
> 
> also yea vm error bad

@KitCat962 IIRC from reading the source that one time, it's the entity's serialized NBT (JSON containing `id` property and other entity properties next to it)

**Comment by KitCat962** [comment]
Also, none of the functions added to EntityTask return self. They should return self to match all other RnederTask functions

**Comment by applejuiceyy** [comment]
> are you sure about that?
> errors with or without the `{}`
> 
> image
> 
> also yea vm error bad

no just the nbt part

**Comment by applejuiceyy** [comment]
> Also, none of the functions added to EntityTask return self. They should return self to match all other RnederTask functions

yes, calm down, I only have 4 hours per day and not all of them are in intellij

**Comment by KitCat962** [comment]
> > Also, none of the functions added to EntityTask return self. They should return self to match all other RnederTask functions
> 
> yes, calm down, I only have 4 hours per day and not all of them are in intellij

sorry. I got too excited. I can't tell what is a "i forgor" and what is a "I know about this and I'll do it later". Usually prs are only put in draft when changes are still being done, so since this was an open pr I assumed what I saw was going to be "final".

File: common/src/main/java/org/figuramc/figura/model/FiguraModelPart.java [hunk 1]
```diff
@@ -1303,6 +1302,19 @@ public SpriteTask newSprite(@LuaNotNil String name) {
         return task;
     }
 
+    @LuaWhitelist
+    @LuaMethodDoc(
+            overloads = @LuaMethodOverload(
+                    argumentTypes = String.class,
+                    argumentNames = "taskName"
+            ),
+            value = "model_part.new_block")
```
Comment (4P5) [inline common/src/main/java/org/figuramc/figura/model/FiguraModelPart.java]: Was `model_part.new_block` accidentally left in?

**Comment by UnlikePaladin** [comment]
lgtm

**Pull request merged**